kloppy/infra/serializers/event/sportec/serializer.py

```diff
@@ -0,0 +1,479 @@
+
+
+class SportecEventSerializer(EventDataSerializer):
+    @staticmethod
+    def __validate_inputs(inputs: Dict[str, Readable]):
+        if "match_data" not in inputs:
+            raise ValueError("Please specify a value for input 'match_data'")
+        if "event_data" not in inputs:
+            raise ValueError("Please specify a value for input 'event_data'")
+
+    def deserialize(
+        self, inputs: Dict[str, Readable], options: Dict = None
+    ) -> EventDataset:
+        self.__validate_inputs(inputs)
+        if not options:
+            options = {}
+
+        with performance_logging("load data", logger=logger):
+            match_root = objectify.fromstring(inputs["match_data"].read())
+            event_root = objectify.fromstring(inputs["event_data"].read())
+
+            wanted_event_types = [
+                EventType[event_type.upper()]
+                for event_type in options.get("event_types", [])
+            ]
+
+        with performance_logging("parse data", logger=logger):
+            x_max = float(
+                match_root.MatchInformation.Environment.attrib["PitchX"]
+            )
+            y_max = float(
+                match_root.MatchInformation.Environment.attrib["PitchY"]
+            )
+
+            team_path = objectify.ObjectPath(
+                "PutDataRequest.MatchInformation.Teams"
+            )
+            team_elms = list(team_path.find(match_root).iterchildren("Team"))
+
+            for team_elm in team_elms:
+                if team_elm.attrib["Role"] == "home":
+                    home_team = _team_from_xml_elm(team_elm)
+                elif team_elm.attrib["Role"] == "guest":
+                    away_team = _team_from_xml_elm(team_elm)
+                else:
+                    raise Exception(f"Unknown side: {team_elm.attrib['Role']}")
+
+            (
+                home_score,
+                away_score,
+            ) = match_root.MatchInformation.General.attrib["Result"].split(":")
+            score = Score(home=int(home_score), away=int(away_score))
+            teams = [home_team, away_team]
+
+            if len(home_team.players) == 0 or len(away_team.players) == 0:
+                raise Exception("LineUp incomplete")
+
+            periods = []
+            period_id = 0
+            events = []
+
+            for event_elm in event_root.iterchildren("Event"):
+                event_chain = _event_chain_from_xml_elm(event_elm)
+                timestamp = _parse_datetime(event_chain["Event"]["EventTime"])
+                if (
+                    SPORTEC_EVENT_NAME_KICKOFF in event_chain
+                    and "GameSection"
+                    in event_chain[SPORTEC_EVENT_NAME_KICKOFF]
+                ):
+                    period_id += 1
+                    period = Period(
+                        id=period_id,
+                        start_timestamp=timestamp,
+                        end_timestamp=None,
```

Comment:
I understand there is no end time for the periods? May be using the start time of the next one as end of the previous one? Or better leave that to the user to patch if needed?

Reply:
The period is ended by the 'FinalWhistle' event. So until an half is ended the period.end_timestamp will be None. But it will be updated .